dump-logs/action.yml

```diff
@@ -0,0 +1,46 @@
+name: "Dump logs"
+description: "Collects all charm and k8s logs"
+
+runs:
+  using: "composite"
+  steps:
+  - name: Dump debug log
+    run: |
+      for ctl in $(juju controllers --format json | jq -r '.controllers | keys[]'); do for mdl in $(juju models --format json | jq -r '.models[].name' | grep -v "admin/controller"); do juju debug-log -m $ctl:$mdl --replay --ms --no-tail; done; done || true
+      exit 0
+    shell: bash
+  - name: Dump pods and their logs
+    shell: bash
+    run: |
+      juju status --relations --storage
+      kubectl get pods \
+          -A \
+          -o=jsonpath='{range.items[*]}{.metadata.namespace} {.metadata.name}{"\n"}' \
+          --sort-by=.metadata.namespace \
+          | grep -v "^\s*$" \
+          | while read namespace pod; do \
+               kubectl -n $namespace describe pod $pod; \
+               kubectl -n $namespace logs $pod \
+                  --all-containers=true \
+                  --tail=100; \
+           done
+  - name: Dump deployments
+    run: |
+      kubectl describe deployments -A
+      exit 0
```

Comment:
Why do you need the explicit `exit 0` here and in the following steps?

Reply:
Since we're at a race with pytest-operator destroying everything, some of these log collection attempts may fail, so `exit 0` to make sure the step "succeeds" so the workflow won't abort before attempting collecting all types of logs.